plugins/pur/core/src/main/java/org/pentaho/di/repository/pur/PurRepository.java

```diff
@@ -2880,11 +3220,17 @@ public JobMeta loadJob( ObjectId idJob, String versionLabel ) throws KettleExcep
   public TransMeta loadTransformation( ObjectId idTransformation, String versionLabel ) throws KettleException {
     try {
       RepositoryFile file = null;
-      if ( versionLabel != null ) {
-        file = pur.getFileAtVersion( idTransformation.getId(), versionLabel );
-      } else {
-        file = pur.getFileById( idTransformation.getId() );
+      readWriteLock.readLock().lock();
+      try {
+        if ( versionLabel != null ) {
+          file = pur.getFileAtVersion( idTransformation.getId(), versionLabel );
+        } else {
+          file = pur.getFileById( idTransformation.getId() );
+        }
+      } finally {
+        readWriteLock.readLock().unlock();
```

Comment:
The read lock should not be release between this and lower in the same method. The lock should be kept.